platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/uptime/UptimeMetrics.java

```diff
@@ -0,0 +1,212 @@
+                        CATEGORY, TIME_SINCE_LAST_CONSENSUS_EVENT + nodeId)
+                .withUnit("seconds")
+                .withDescription("The consensus time in seconds since the "
+                        + "last consensus event created by this node was observed");
+        timeSinceLastConsensusEvent.put(nodeId, metrics.getOrCreate(timeSinceLastConensusEventConfig));
+
+        final RunningAverageMetric.Config roundsSinceLastConensusEventConfig = new RunningAverageMetric.Config(
+                        CATEGORY, ROUNDS_SINCE_LAST_CONSENSUS_EVENT + nodeId)
+                .withUnit("rounds")
+                .withDescription(
+                        "The number of rounds since the " + "last consensus event created by this node was observed");
+        roundsSinceLastConsensusEvent.put(nodeId, metrics.getOrCreate(roundsSinceLastConensusEventConfig));
+
+        final RunningAverageMetric.Config timeSinceLastJudgeConfig = new RunningAverageMetric.Config(
+                        CATEGORY, TIME_SINCE_LAST_JUDGE + nodeId)
+                .withUnit("seconds")
+                .withDescription(
+                        "The consensus time in seconds since the " + "last judge created by this node was observed");
+        timeSinceLastJudge.put(nodeId, metrics.getOrCreate(timeSinceLastJudgeConfig));
+
+        final RunningAverageMetric.Config roundsSinceLastJudgeConfig = new RunningAverageMetric.Config(
+                        CATEGORY, ROUNDS_SINCE_LAST_JUDGE + nodeId)
+                .withUnit("rounds")
+                .withDescription("The number of rounds since the " + "last judge created by this node was observed");
+        roundsSinceLastJudge.put(nodeId, metrics.getOrCreate(roundsSinceLastJudgeConfig));
+    }
+
+    /**
+     * Remove the metrics for a node.
+     *
+     * @param nodeId the id of the node
+     */
+    public void removeMetricsForNode(final long nodeId) {
+        timeSinceLastConsensusEvent.remove(nodeId);
+        metrics.remove(new RunningAverageMetric.Config(CATEGORY, TIME_SINCE_LAST_CONSENSUS_EVENT + nodeId));
+
+        roundsSinceLastConsensusEvent.remove(nodeId);
+        metrics.remove(new RunningAverageMetric.Config(CATEGORY, ROUNDS_SINCE_LAST_CONSENSUS_EVENT + nodeId));
+
+        timeSinceLastJudge.remove(nodeId);
+        metrics.remove(new RunningAverageMetric.Config(CATEGORY, TIME_SINCE_LAST_JUDGE + nodeId));
+
+        roundsSinceLastJudge.remove(nodeId);
+        metrics.remove(new RunningAverageMetric.Config(CATEGORY, ROUNDS_SINCE_LAST_JUDGE + nodeId));
+    }
+
+    /**
+     * Get the metric that tracks the time since the last consensus event was observed from a node.
+     *
+     * @param id the id of the node
+     * @return the metric
+     */
+    public @NonNull RunningAverageMetric getTimeSinceLastConsensusEventMetric(final long id) {
+        return Objects.requireNonNull(timeSinceLastConsensusEvent.get(id));
```

Comment:
I agree this is the wrong exception, probably `NoSuchElementException` is better. Changed.

In my opinion, it makes for cleaner API to throw an exception if the user does something silly, rather than requiring the user to do a null check after the call.